processor/lsmintervalprocessor/internal/merger/model.go

```diff
@@ -406,3 +417,126 @@ func mergeDeltaHistogramDP(from, to pmetric.HistogramDataPoint) {
+		to.SetMin(min(to.Min(), from.Min()))
+		to.SetMax(max(to.Max(), from.Max()))
+	}
+
+	to.SetSum(to.Sum() + from.Sum())
+	to.SetCount(to.Count() + from.Count())
+	// TODO (lahsivjar): Zero count depends on the value of zero threshold.
+	// If zero threshold are not the same then we can't merge them directly,
+	// instead, we will need to expand the zero threshold to be inclusive
+	// of the smaller zero count. If a zero threshold encroaches a bucket,
+	// then we will need to expand the zero count till the end of the bucket.
+	to.SetZeroCount(to.ZeroCount() + from.ZeroCount())
+
+	// Downscale the resolution and use perfect-subsetting
+	minScale := min(to.Scale(), from.Scale())
+	maybeScaleDownHistogram(to, minScale)
+	maybeScaleDownHistogram(from, minScale)
+
+	// All histogram buckets are now on same scale, merge them
+	mergeExponentialHistogramBuckets(from.Positive(), to.Positive())
+	mergeExponentialHistogramBuckets(from.Negative(), to.Negative())
+}
+
+func mergeExponentialHistogramBuckets(from, to pmetric.ExponentialHistogramDataPointBuckets) {
+	fromBuckets := from.BucketCounts()
+	toBuckets := to.BucketCounts()
+	fromSize := int(from.Offset()) + fromBuckets.Len()
+	toSize := int(to.Offset()) + toBuckets.Len()
+
+	if to.Offset() > from.Offset() {
+		// Expand `to` to accomodate the merged buckets with offset as `from.Offset()`
+		newBucketsSize := max(fromSize, toSize) - int(from.Offset())
+		newBuckets := pcommon.NewUInt64Slice()
+		newBuckets.Append(make([]uint64, newBucketsSize)...)
+
+		// Copy `to` to the new bucket
+		for i := 0; i < toBuckets.Len(); i++ {
+			newBuckets.SetAt(int(to.Offset())+i, toBuckets.At(i))
+		}
+		newBuckets.MoveTo(to.BucketCounts())
+		to.SetOffset(from.Offset())
+
+		toBuckets = to.BucketCounts()
+		toSize = int(to.Offset()) + toBuckets.Len()
+	} else if fromSize > toSize {
+		toBuckets.Append(make([]uint64, fromSize-toSize)...)
+	}
+	for i := 0; i < fromBuckets.Len(); i++ {
+		toOffset := int(from.Offset()-to.Offset()) + i
+		toBuckets.SetAt(toOffset, toBuckets.At(toOffset)+fromBuckets.At(i))
+	}
+}
+
+// maybeScaleDownHistogram attempts to scale down histgram if the proper scale is passed.
+func maybeScaleDownHistogram(hist pmetric.ExponentialHistogramDataPoint, newScale int32) {
+	diff := hist.Scale() - newScale
+	switch {
+	case diff < 0:
+		panic("unexpected state, histogram cannot be upscaled")
+	case diff == 0:
+		return
+	}
+
+	scaleDownHistogramBuckets(hist.Positive(), diff)
+	scaleDownHistogramBuckets(hist.Negative(), diff)
+}
+
+// scaleDownHistogramBuckets scales down the histogram buckets by collapsing 2^(scaleDiff)
+// number of buckets into one bucket while handling the offset as required.
+// TODO (lahsivjar): Currently we don't trim the trailing zero entries from the buckets.
+// This is because OTel doesn't expose the data-model to efficiently slice the buckets,
+// instead, trimming trailing zero entries will require reallocation of the whole slice.
+// Is it worth it? (See test cases)
```

Comment:
[**For reviewers**] I don't think this is a big issue (keeping trailing zeroes) but let me know if there are any concerns.

Reply:
I wouldn't worry about it for now.

Reply:
👍 For posterity (and for me as well when I read this back), this is easy to do by just doing right-shift on the effective len and reslicing. The only reason to not do is due to the extra reallocation involved. Sample code:

```
	end := (hist.BucketCounts().Len() + int(hist.Offset()) - 1) >> scaleDiff // using offset before scaledown
	hist.BucketCounts().FromRaw(hist.BucketCounts().AsRaw()[:(end + 1)])
```